### PR TITLE
added working bookstack-traefik installation

### PR DESCRIPTION
**Pull request by xFuture603**
This PR contains a working state of a bookstack installation, using Traefik as a reverse proxy. We use docker-compose to set up the whole thing.

On top we add renovate to keep track of new software versions to update our compose file accordingly.

### CODE REVIEW

File: README.md
Comment (tuhlmann) [inline README.md:28]: Is mail.com a reserved example domain, like example.com, or could this end up in a real inbox?

File: README.md
Comment (tuhlmann) [inline README.md:30]: How does the user create their encrypted basic auth string. Should that described here? What's the current value of this string? Wouldn't it be safer to put no value at all in. That way the lazy user cannot just use the password and opens up a security hole, but is forced to create their own password?

File: traefik-letsencrypt-bookstack-docker-compose.yml
Comment (tuhlmann) [inline traefik-letsencrypt-bookstack-docker-compose.yml:51]: Is there an alternative to putting password in the compose file?

**Review by tuhlmann — CHANGES_REQUESTED** [review]
Asked a few questions I'd like to have resolved before approval.

File: README.md
Comment (xFuture603) [reply README.md]: I've updated the example mail to something non-existent. Thanks for the hint.

File: README.md
Comment (xFuture603) [reply README.md:30]: I've added a documentation part on how you can encrypt a password to use it in your basic auth.

File: traefik-letsencrypt-bookstack-docker-compose.yml
Comment (xFuture603) [reply traefik-letsencrypt-bookstack-docker-compose.yml:51]: Sure, you can add a .env file for example or you can vault the hashed password on top with Ansible for example. In my opinion it is okay to leave the hashed password in the compose file because it is already encrypted.

File: traefik-letsencrypt-bookstack-docker-compose.yml
Comment (tuhlmann) [inline traefik-letsencrypt-bookstack-docker-compose.yml:51]: Sorry to bother you, but it doesn't feel right to me to have passwords- even encrypted ones- in a docker-compose. The reasons might be subtle. I view docker-compose (or any other script) as a rather stable receipt the computer can execute. It shouldn't be touched if a password needs to be updated.
If you could add a `.env.example` file with the key for the password entry in it and a comment but no value, and then instructions how to copy and create that password. I want
- the user having to take action and create their own password, not take a public one because their lazy
- move the password out of the compose and just have compose reference the .env key. Now, I have no idea what you have to do to make compose read passwords from env- I hope that's somewhat straight forward? But you suggested that approach :)

**Review by tuhlmann — CHANGES_REQUESTED** [review]
Just one issue left I have with this password in docker-compose. Can you address my comment? If you feel strongly that you shouldn't let's chat about it...

File: traefik-letsencrypt-bookstack-docker-compose.yml
Comment (xFuture603) [reply traefik-letsencrypt-bookstack-docker-compose.yml]: Sorry for the delay in this PR, due to holidays.

I've added a .env file and moved the basic_auth user and password to it. I've added a unencoded string to the password variable (won't work without encoding to bcrypt for example).

Now you have variables that in the docker-compose file that reference to the .env variables. I hope that everything is fine now. :)

**Review by tuhlmann — APPROVED** [review]
Thank you for addressing these issues!!

I have a few comments for future improvements:
- Add a note to the docker-compose comment about which encryption alogorithms can be used- add that BCrypt is strongly recommended
- We normally use upper case (and separated by underscore) environment variable. I'm not sure how this is handled in the Docker/Ansible world. I'm fine with a different standard, but if the same convention applies to Docker/Ansible I suggest with switch out .env variables to UPPER_CASE as well.

**Comment by xFuture603** [comment]
> Thank you for addressing these issues!!
> 
> I have a few comments for future improvements:
> 
> * Add a note to the docker-compose comment about which encryption alogorithms can be used- add that BCrypt is strongly recommended
> * We normally use upper case (and separated by underscore) environment variable. I'm not sure how this is handled in the Docker/Ansible world. I'm fine with a different standard, but if the same convention applies to Docker/Ansible I suggest with switch out .env variables to UPPER_CASE as well.

I've addressed both issues and added them to the PR. I will merge this PR now.

**Pull request merged**
